main_test.go

```diff
@@ -375,10 +383,10 @@ func optionsFromOSARCH(osarch string, sema chan struct{}) compileopts.Options {
 }
 
 func runTest(name string, options compileopts.Options, t *testing.T, cmdArgs, environmentVars []string) {
-	runTestWithConfig(name, t, options, cmdArgs, environmentVars)
+	runTestWithConfig(name, t, options, cmdArgs, environmentVars, "")
 }
 
-func runTestWithConfig(name string, t *testing.T, options compileopts.Options, cmdArgs, environmentVars []string) {
+func runTestWithConfig(name string, t *testing.T, options compileopts.Options, cmdArgs, environmentVars []string, emulator string) {
```

Comment:
Rather than override `emulator`, what if struct `TargetSpec` had an `EmulatorArgs` field that is injected into the argument string for the default emulator for a target?

It could be injected in `(*Config).Emulator()`, appending `" " + c.Target.EmulatorArgs`.

Reply:
Fixed this by removing the new test entirely and replacing it with a much more extensive test using wazero (one that I'm actually confident in will work).